src/emqx_bridge.erl

```diff
@@ -91,6 +113,48 @@ handle_call(status, _From, State = #state{client_pid = undefined}) ->
 handle_call(status, _From, State = #state{client_pid = _Pid})->
     {reply, <<"Running">>, State};
 
+handle_call(show_forwards, _From, State = #state{forwards = Forwards}) ->
+    {reply, Forwards, State};
+
+handle_call({add_forward, Topic}, _From, State = #state{forwards = Forwards}) ->
+    case emqx_topic:validate({filter, Topic}) andalso (not lists:member(Topic, Forwards)) of
```

Comment:
emqx_topic:validate/1 is side-effect free except for error exceptions.
the exceptions may cause gen_server to crash, move validation to the gen_server:call wrapper ?

Reply:
Agree:)